pubsub/cloud-client/subscriptions.py

```diff
@@ -0,0 +1,297 @@
+
+
+# [START pubsub_list_subscriptions]
+def list_subscriptions():
+    # Lists all subscriptions in the current project
+    subscriptions = []
+    next_page_token = None
+    while True:
+        page, next_page_token = pubsub_client.list_subscriptions()
+        subscriptions.extend(page)
+        if not next_page_token:
+            break
+
+    print('Subscriptions:')
+    for subscription in subscriptions:
+        print(subscription.name)
+# [END pubsub_list_subscriptions]
+
+
+# [START pubsub_list_topic_subscriptions]
+def list_topic_subscriptions(topic_name):
+    # Reference an existing topic, e.g. "my-topic"
+    topic = pubsub_client.topic(topic_name)
+
+    # Lists all subscriptions for the topic
+    subscriptions = []
+    next_page_token = None
+    while True:
+        page, next_page_token = topic.list_subscriptions()
+        subscriptions.extend(page)
+        if not next_page_token:
+            break
+
+    print('Subscriptions:')
+    for subscription in subscriptions:
+        print(subscription.name)
+# [END pubsub_list_topic_subscriptions]
+
+
+# [START pubsub_create_subscription]
+def create_subscription(topic_name, subscription_name):
+    # References an existing topic, e.g. "my-topic"
+    topic = pubsub_client.topic(topic_name)
+
+    # Prepares a new subscription, e.g. "my-new-subscription"
+    subscription = topic.subscription(subscription_name)
+
+    # Creates the subscription
+    subscription.create()
+
+    print('Subscription {} created.'.format(subscription.name))
+# [END pubsub_create_subscription]
+
+
+# [START pubsub_create_push_subscription]
+def create_push_subscription(topic_name, subscription_name):
+    # References an existing topic, e.g. "my-topic"
+    topic = pubsub_client.topic(topic_name)
+
+    project_id = os.environ['GCLOUD_PROJECT'] or 'YOUR_PROJECT_ID'
+
+    # Prepares a new push subscription, e.g. "my-new-subscription"
+    subscription = topic.subscription(
+        subscription_name,
+        # Set to an HTTPS endpoint of your choice. If necessary, register
+        # (authorize) the domain on which the server is hosted.
+        push_endpoint = 'https://{}.appspot.com/push'.format(project_id)
+    )
+
+    # Creates the push subscription
+    subscription.create()
+
+    print('Subscription {} created.'.format(subscription.name))
+# [END pubsub_create_push_subscription]
+
+
+# [START pubsub_delete_subscription]
+def delete_subscription(topic_name, subscription_name):
+    # References an existing topic, e.g. "my-topic"
+    topic = pubsub_client.topic(topic_name)
+
+    # References an existing subscription, e.g. "my-subscription"
+    subscription = topic.subscription(subscription_name)
+
+    # Deletes the subscription
+    subscription.delete()
+
+    print('Subscription {} deleted.'.format(subscription.name))
+# [END pubsub_delete_subscription]
+
+
+# [START pubsub_get_subscription_metadata]
+def get_subscription_metadata(topic_name, subscription_name):
```

Comment:
`get_subscription` is just a good as `get_subscription_metadata` and is more succinct.